ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -244,94 +244,66 @@ def empty(spec: "BehaviorSpec") -> "TerminalSteps":
         """
-        return self.action_type == ActionType.DISCRETE
+        return self.discrete_size > 0 and self.continuous_size == 0
 
-    def is_action_continuous(self) -> bool:
+    # For backwards compatibility
+    def is_continuous(self) -> bool:
         """
         Returns true if this Behavior uses continuous actions
         """
-        return self.action_type == ActionType.CONTINUOUS
+        return self.discrete_size == 0 and self.continuous_size > 0
 
     @property
-    def action_size(self) -> int:
-        """
-        Returns the dimension of the action.
-         - In the continuous case, will return the number of continuous actions.
-         - In the (multi-)discrete case, will return the number of action.
-         branches.
-        """
-        if self.action_type == ActionType.DISCRETE:
-            return len(self.action_shape)  # type: ignore
-        else:
-            return self.action_shape  # type: ignore
+    def discrete_branches(self) -> Tuple[int, ...]:
+        return self.discrete_branch_sizes  # type: ignore
 
     @property
-    def discrete_action_branches(self) -> Optional[Tuple[int, ...]]:
-        """
-        Returns a Tuple of int corresponding to the number of possible actions
-        for each branch (only for discrete actions). Will return None in
-        for continuous actions.
-        """
-        if self.action_type == ActionType.DISCRETE:
-            return self.action_shape  # type: ignore
-        else:
-            return None
+    def discrete_size(self) -> int:
+        return len(self.discrete_branch_sizes)
 
-    def create_empty_action(self, n_agents: int) -> np.ndarray:
-        """
-        Generates a numpy array corresponding to an empty action (all zeros)
-        for a number of agents.
-        :param n_agents: The number of agents that will have actions generated
-        """
-        if self.action_type == ActionType.DISCRETE:
-            return np.zeros((n_agents, self.action_size), dtype=np.int32)
-        else:
-            return np.zeros((n_agents, self.action_size), dtype=np.float32)
+    @property
+    def continuous_size(self) -> int:
+        return self.num_continuous_actions
 
-    def create_random_action(self, n_agents: int) -> np.ndarray:
-        """
-        Generates a numpy array corresponding to a random action (either discrete
-        or continuous) for a number of agents.
-        :param n_agents: The number of agents that will have actions generated
-        :param generator: The random number generator used for creating random action
-        """
-        if self.is_action_continuous():
+    def create_empty(self, n_agents: int) -> np.ndarray:
+        if self.is_continuous():
+            return np.zeros((n_agents, self.continuous_size), dtype=np.float32)
+        return np.zeros((n_agents, self.discrete_size), dtype=np.int32)
+
+    def create_random(self, n_agents: int) -> np.ndarray:
```

Comment:
docstring

Reply:
I think we should keep _action is there (because we are not creating an action_spec!). Also, since we are renaming these, I don't like `create` anymore. Maybe `generate` or `make` to be consistent with `make_continuous` ?